src/main/resources/xslt/preservica2solr.xsl

```diff
@@ -0,0 +1,178 @@
+            </f:string>
+          </xsl:if>
+        </xsl:for-each>
+
+        <!--TODO: Figure out what the difference is between kanalnavn and channel_name in the metadata. -->
+        <xsl:for-each select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcorePublisher">
+          <xsl:if test="publisherRole = 'kanalnavn'">
+            <f:string key="creator_affiliation">
+              <xsl:value-of select="publisher"/>
+            </f:string>
+          </xsl:if>
+        </xsl:for-each>
+
+        <f:string key="notes">
+          <xsl:value-of select="normalize-space(/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreDescription/description)"/>
+        </f:string>
+
+        <!-- Video specific transformations -->
+        <xsl:for-each select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreIdentifier">
+          <xsl:if test="identifierSource = 'ritzauId'">
+            <f:string key="ritzau_id">
+              <xsl:value-of select="identifier"/>
+            </f:string>
+          </xsl:if>
+          
+          <xsl:if test="identifierSource = 'tvmeterId'">
+            <f:string key="tvmeter_id">
+              <xsl:value-of select="identifier"/>
+            </f:string>
+          </xsl:if>
+        </xsl:for-each>
+
+        <!--Some docs have this duration field. Others doesn't. Then we need to extract the duration from pbcoreDateAvailable-->
+        <xsl:choose>
+          <xsl:when test="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/formatDuration">
+            <f:string key="duration">
+              <xsl:value-of select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/formatDuration"/>
+            </f:string>
+          </xsl:when>
+          <xsl:otherwise>
+
+            <xsl:variable name="startTime">
+              <xsl:value-of select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/pbcoreDateAvailable/dateAvailableStart"/>
+            </xsl:variable>
+            <xsl:variable name="endTime">
+              <xsl:value-of select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/pbcoreDateAvailable/dateAvailableEnd"/>
+            </xsl:variable>
+            <xsl:variable name="durationInMilliseconds">
+              <xsl:value-of select="my:toMilliseconds($startTime, $endTime)"/>
+            </xsl:variable>
+
+            <f:string key="duration">
+              <xsl:value-of select="$durationInMilliseconds"/>
+            </f:string>
+          </xsl:otherwise>
+        </xsl:choose>
+
+
+
+        <xsl:if test="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/formatColors = 'farve'">
+          <f:string key="color">true</f:string>
+        </xsl:if>
+        <xsl:if test="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument/pbcoreInstantiation/formatColors != 'farve'">
```

Comment:
https://stackoverflow.com/questions/13622338/how-to-implement-if-else-statement-in-xslt :-)